src/core/solvers/ddp.cpp

```diff
@@ -139,16 +139,7 @@ double SolverDDP::tryStep(const double steplength) {
 }
 
 double SolverDDP::stoppingCriteria() {
-  stop_ = 0.;
-  const std::size_t T = this->problem_->get_T();
-  const std::vector<boost::shared_ptr<ActionModelAbstract> >& models = problem_->get_runningModels();
-
-  for (std::size_t t = 0; t < T; ++t) {
-    const std::size_t nu = models[t]->get_nu();
-    if (nu != 0) {
-      stop_ += Qu_[t].head(nu).squaredNorm();
-    }
-  }
+  stop_ = abs(d_[0]);
```

Comment:
why do you need an abs here?

Reply:
We don't need the abs here, d_[0] would remain positive by definition.

Reply:
Could you please remove it?

Reply:
This might be indeed a possible choice, despite that the `abs` operation doesn't hurt, but I am not fully sure now.
Please give me some time to come with a definite conclusion.

Side comment:
The `abs` operation is important in equality-constrained cases whether d[0] can be negative. I found often in these cases that very tiny negative values (appr. -1e-12) at the local optimum.

Reply:
Another comment (sorry for multiple responses):
In equality-constrained cases, d[0] might compute an ascent direction (i.e., a substantially big negative value), which produces an undesired (silly) early stopping procedure as we check that `d[0] < th_stop`.

Reply:
In that case, this is something that belongs to the specific solver that you are working with. 
I see no reason to change the stopping criteria in base ddp class.

Reply:
> In that case, this is something that belongs to the specific solver that you are working with.
> I see no reason to change the stopping criteria in base ddp class.

Yes, I agree with that. To be honest, my main concern is with the FDDP, where we know we have (and can accept) ascent directions. So, if this is case we might need to re-implement the `stoppingCriteria` function.

Reply:
For FDDP, we might handle separately this issue then, once we understand it better. 
I believe for your solver, implementing in the solver itself would be the best way forward. 
I'll close this PR following the discussion

Reply:
Hey @proyan -- I understand well the need of using dV[0] as stopping criteria, please do not think the opposite. But you asked me to remove `abs` operation, and for this last bit, I need to think first.

The reason why I didn't check it first is due to the fact that the `abs` operation won't hurt (in any form) the measurement of the stopping criteria. And to be honest with you, I think we could go ahead as this doesn't really matters. What matters is to discuss `dV[0]` vs `||Qu||`, despite that I am convinced on my arguments and empirical evidence.

Therefore, I'll reopen this issue based on above explanations :)